### ISSUE TITLE
SafeAreaProvider does not trigger re-render with updated theme

### ISSUE THREAD

**Issue by whydidoo** [jpudysz/react-native-unistyles#1018]
### What happened?

When wrapping components with <SafeAreaProvider> and applying a theme using UnistylesRuntime.setTheme, the children do not re-render automatically to reflect theme changes.

https://github.com/user-attachments/assets/ca989131-dec1-412f-a91d-33af78307a41

### Steps to Reproduce

1) Use the following component:
```
export const ModuleProvidersWrapper = () => {
  useEffect(() => {
    UnistylesRuntime.setTheme('dark');
  }, []);

  return (
    <SafeAreaProvider>
      <View style={styles.view}>
        <Text style={styles.text({ color: 'Text.Primary' })}>
          BGCOLOR MUST BE DARK, TEXT MUST BE LIGHT
        </Text>
      </View>
    </SafeAreaProvider>
  );
};

const styles = StyleSheet.create((theme, rt) => {
  console.log(
    `Mode: ${theme.mode}, theme.palette.Text.Primary: ${theme.palette.Text.Primary}`,
    `Mode: ${theme.mode}, theme.palette.Background.Paper: ${theme.palette.Background.Paper}`,
  );

  return {
    text: ({ color }: { color: string }) => ({
      color: get(theme.palette, color) ?? color,
    }),
    view: {
      paddingTop: rt.insets.top,
      flex: 1,
      backgroundColor: theme.palette.Background.Paper,
    },
  };
});

```

2) Set the theme via UnistylesRuntime.setTheme('dark')

3) Observe that **styles.view** and **styles.text** do not update with the expected dark background and light text.

**Expected behavior:**
Background color should be dark.
Text color should be light.
Components should automatically re-render when the theme is applied or updated.

**Actual behavior:**
The initial render uses the light theme.
<View> and <Text> do not update even after the theme is applied.

**Notes:**
SafeAreaProvider itself does not trigger a re-render when the theme changes for the first time.
If SafeAreaProvider is not present in the DOM initially, the theme updates work as expected.

### Unistyles Version

3.0.15

### React Native Version

0.81

### Platforms

iOS

### Expo

None

### Additional information

- [x] I use recommended version of `react-native-nitro-modules` [compatibility table](https://github.com/jpudysz/react-native-unistyles?tab=readme-ov-file#installation)
- [x] I've searched for [similar issues in this repository](https://github.com/jpudysz/react-native-unistyles/issues) and found none
- [x] I double-checked that my issue was not covered in [documentation](https://unistyl.es/v3/guides/why-my-view-doesnt-update)

**Comment by whydidoo** [comment]
@jpudysz Small update — when I used `styles.useVariants({})`, the theme was updating in the components. 

For example my component

```
export function Typography<T extends ExnessTypographyVariants>(
  props: TextProps<T> & React.RefAttributes<Text>
) {
  const {
    style,
    variant = 'Body.B1_Regular',
    color = 'Text.Primary',
    ...otherProps
  } = props;

  stylesheet.useVariants({ variant });

  return (
    <Text
      style={[stylesheet.root, stylesheet.text({ color, variant }), style]}
      role={ROLE[variant]}
      aria-level={variant ? ARIA_LEVEL[variant] : undefined}
      {...otherProps}
    />
  );
}

const stylesheet = StyleSheet.create((theme, rt) => ({
  root: {
    variants: {
      variant: theme.typography,
    }
  },
  text: ({
    color,
    variant,
  }: {
    color: ColorsTypePath;
    variant: ExnessTypographyVariants;
  }) => {
    const colorValue = get(theme.palette, color) ?? color;

    return {
      color: colorValue,
      writingDirection: Platform.select({
        web: undefined,
        default: rt.rtl ? 'rtl' : 'ltr',
      }),
    };
  },
}));

```

**Comment by jpudysz** [comment]
Wrapping your app in `SafeAreaProvider` will conditionally render app:

https://github.com/AppAndFlow/react-native-safe-area-context/blob/8b2267a4726b7520e82d35f26d23bacce9ec44c9/src/SafeAreaContext.tsx#L97

At the beginning `insets` are null:

<img width="542" height="136" alt="Image" src="https://github.com/user-attachments/assets/847d7ba2-be33-4d7a-9cbb-3ba6900bf3e5" />

So `useEffect` runs before `View` and `Text` are mounted. 
The question is - why `View` and `Text` have invalid colors after mount. Checking it now.

**Comment by jpudysz** [comment]
This is a race condition:

1. `View` and `Text` are computed with the light theme because React keeps them as `children` in the `SafeAreaContext` prop, so they’re considered rendered
2. Since there are no insets, `View` and `Text` are not mounted (not present in the `ShadowTree`)
3. `useEffect` fires with the theme change event, but there are no nodes that could be affected (the `ShadowTree` is empty)  
4. Because React stored `View` and `Text` as children, their underlying logic is no longer called, they don't try to get styles and they are just rendered as-is.

This is technically a bug, but that’s how React works. I can’t affect these nodes anymore because they’ve already been rendered, even though they’re not present when the theme changes.

It just needs to be executed once the `SafeAreaContext` resolves the insets.

Few options to fix it.

1. User `requestIdleCallback` or `setTimeout` to delay the theme change
```ts
useEffect(() => {
  requestIdleCallback(() => {
    UnistylesRuntime.setTheme('dark');
  });
}, []);
```
2. Use separate component so React won't "cache it":

```ts
const SeparateComponent = () => {
    return (
        <View style={styles.view}>
        <Text style={styles.text({ color: 'aloes' })}>
          BGCOLOR MUST BE DARK, TEXT MUST BE LIGHT
        </Text>
    </View>
    )
}

export const App = () => {
  useEffect(() => {
    UnistylesRuntime.setTheme('dark');
  }, []);

  return (
    <SafeAreaProvider>
        <SeparateComponent />
    </SafeAreaProvider>
  );

}
```

3. Use react-native-unistyles insets

```tsx
export const App = () => {
  useEffect(() => {
    UnistylesRuntime.setTheme('dark');
  }, []);

  return (
    <View style={styles.view}>
        <Text style={styles.text({ color: 'aloes' })}>
          BGCOLOR MUST BE DARK, TEXT MUST BE LIGHT
        </Text>
    </View>
  );

}

const styles = StyleSheet.create((theme, rt) => ({
    text: ({ color }: { color: keyof typeof theme.colors }) => ({
      color: theme.colors[color]
    }),
    view: {
      paddingTop: rt.insets.top,
      flex: 1,
      backgroundColor: theme.colors.backgroundColor,
    },
}))
```

**Comment by whydidoo** [comment]
@jpudysz, thank you for conducting the investigation of the issue. I will take the identified limitations into account moving forward.

**Comment by jpudysz** [comment]
Yeah, no problem. Sorry I don't have the fix, but this is super rare setup. It's just a matter of timing and React internals.
